js/ui/scheduler/workspaces/virtual_selection_state.js

```diff
@@ -111,15 +118,31 @@ export default class VirtualSelectionState {
         return groupIndex === nextGroupIndex && allDay === nextAllDay;
     }
 
-    _filterCellsByDate(cellsRow, firstDate, lastDate) {
-        const firstTime = firstDate.getTime();
-        const lastTime = lastDate.getTime();
+    _filterCellsByDateAndIndex(cellsRow, firstDate, lastDate, firstIndex, lastIndex) {
+        const firstDay = dateUtils.trimTime(firstDate).getTime();
+        const lastDay = dateUtils.trimTime(lastDate).getTime();
 
         return cellsRow.filter((cell) => {
-            const { startDate } = cell;
-            const time = startDate.getTime();
+            const { startDate, index } = cell;
+            const day = dateUtils.trimTime(startDate).getTime();
 
-            return firstTime <= time && time <= lastTime;
+            return this._compareCellsByDateAndIndex(day, index, firstDay, firstIndex, lastDay, lastIndex);
         });
     }
+
+    _compareCellsByDateAndIndex(day, index, firstDay, firstIndex, lastDay, lastIndex) {
+        if(firstDay === lastDay) {
+            let _firstIndex = firstIndex;
+            let _lastIndex = lastIndex;
+            if(_firstIndex > _lastIndex) {
+                [_firstIndex, _lastIndex] = [_lastIndex, _firstIndex];
```

Comment:
Usually, we do not use such names. i would suggest to use `validFirstIndex` and `validLastIndex`